File: div1+2/1001/C.cpp
#include<bits/stdc++.h>
#define int long long
using namespace std;

void solve() {
    int n; cin >> n;
    vector<int> a(n);
    for (int i=0; i<n; i++) cin >> a[i];
    int ans = accumulate(a.begin(), a.end(), 0);
    while (a.size() > 1) {
        vector<int> b(a.size()-1);
        for (int j=1; j<a.size(); j++)
            b[j-1] = a[j] - a[j-1];
        ans = max(ans, abs(a.back() - a.front()));
        a = b;
    }
    cout << ans << "\n";
}

int32_t main() {
    ios_base::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    int t; cin >> t;
    while(t--) solve();
}

File: div1+2/1001/C.py
import sys
input = sys.stdin.readline
for i in range(int(input())):
    n = int(input())
    x = list(map(int, input().split()))
    ans = sum(x)
    for i in range(1, n):
        y = [0]*(len(x)-1)
        for j in range(1, len(x)):
            y[j-1] = x[j] - x[j-1] 
        ans = max(abs(x[-1] - x[0]), ans)
        x = y
    print(ans)